pypesto/visualize/parameters.py

```diff
@@ -1,64 +1,89 @@
 import matplotlib.pyplot as plt
+import matplotlib.axes
 from matplotlib.ticker import MaxNLocator
 import numpy as np
 import numbers
-from .reference_points import create_references
+
+from typing import Iterable, List, Optional, Sequence, Tuple, Union
+
+from ..result import Result
+from .reference_points import create_references, ReferencePoint
 from .clust_color import assign_colors
 from .clust_color import delete_nan_inf
 from .misc import process_result_list
 
 
-def parameters(results, ax=None, free_indices_only=True, lb=None, ub=None,
-               size=None, reference=None, colors=None, legends=None,
-               balance_alpha=True, start_indices=None):
+def parameters(
+        results: Union[Result, Sequence[Result]],
+        ax: Optional[matplotlib.axes.Axes] = None,
+        parameter_indices: Union[str, Sequence[int]] = 'free_only',
```

Comment:
```suggestion
        parameter_indices: Union[Literal['all','free_only'], Sequence[int]] = 'free_only',
```

Literal can be imported from `typing`, see https://www.python.org/dev/peps/pep-0586/.

Reply:
But only starting python 3.8, right?

Reply:
Looks like. Currently, pyPESTO is >=3.6. We could move to 3.8, but as long as e.g. anaconda still defaults to 3.7, I would not be too restrictive.

Reply:
Good point, probably shouldn't be changed yet to keep backwards compatibility.